lib/internal/modules/esm/translators.js

```diff
@@ -33,6 +35,19 @@ const debug = debuglog('esm');
 const translators = new SafeMap();
 exports.translators = translators;
 
+const DATA_URL_PATTERN = /^[^/]+\/[^,;]+(;base64)?,([\s\S]*)$/;
+function getSource(url) {
+  const parsed = new URL(url);
+  if (parsed.protocol === 'file:') {
+    return readFileAsync(parsed);
+  } else if (parsed.protocol === 'data:') {
+    const [ , base64, body ] = DATA_URL_PATTERN.exec(parsed.pathname);
```

Comment:
What about the case where the regex here fails parsing? (eg `data:/`)

Reply:
This would error, that invalid URL would have to be returned by a loader though and would already be error checked presumably by the attempted parse in 
 https://github.com/nodejs/node/pull/28614/files/95eab8ae6b9f5c13dbba10b326ba0e41c12b2274#diff-a7c0a5f3e4fc8503fefebbe82071bc38R48

Reply:
But if the `.exec` returns null, the destructuring will throw I think?

Reply:
i'll add some checks

Reply:
fixed in df4dab6